pennylane/interfaces/jax_jit.py

```diff
@@ -251,12 +253,24 @@ def wrapper(p):
             # On the forward execution return the jacobian too
             return res, jacs
 
-        fwd_shapes = [jax.ShapeDtypeStruct((1,), dtype) for _ in range(total_size)]
-        jacobian_shape = [jax.ShapeDtypeStruct((1, len(p)), dtype) for p in params]
+        fwd_shape_dtype_struct = _extract_shape_dtype_structs(tapes, device)
+
+        jacobian_shape = [t.shape(device) + (len(p),) for t in tapes for p in params]
+        jac_dtypes = [_numeric_type_to_dtype(t.numeric_type) for t in tapes]
+
+        # Note: for qml.probs we'll first have a [1,dim] shape for the tape
+        # which is then reduced by the QNode
+        jacobian_shape = [
+            jax.ShapeDtypeStruct(tuple([shape]), dtype)
+            if isinstance(shape, int)
+            else jax.ShapeDtypeStruct(tuple(shape), dtype)
+            for shape, dtype in zip(jacobian_shape, jac_dtypes)

```

Comment:
In what kind of cases would you have `jacobian_shape`, `jac_dtypes` with different length? It appears to me that it should always have `len(tapes)`as length

Reply:
The definition
```py
        jacobian_shape = [t.shape(device) + (len(p),) for t in tapes for p in params]
        jac_dtypes = [_numeric_type_to_dtype(t.numeric_type) for t in tapes]
```
may define `jacobian_shape` and `jac_dtypes` in a way that `len(jacobian_shape) > len(jac_dtypes)`. Frankly, I haven't yet come across cases where this creates an issue, but theoretically, this should be one.

I've marked this as a todo for myself, shouldn't block the other parts of the addition though.

Reply:
Thank you it is a good point to mention 👍